### PR TITLE
updates getChallenges function

### LINKED ISSUE #18: Update getChallenges function
Currently, the getChallenges function returns just the details of a challenge.

We want to extend this to return the tags attached to the challenge too.

### PR DESCRIPTION
**Pull request by mk4111**
Closes #18 

Updates related unit test and server-inject test.
Updates getChallenges.sql file with command to retrieve challenge details and associated tags
Updates initialise_table.sql file to stop it from deleting prepopulated data that the test relies on. ( #16 )
Removes fixed IDs from challenges.json file

### CODE REVIEW

**Comment by mk4111** [comment]
Test failing - looking into this now!

**Comment by mk4111** [comment]
@SimonLab and @JMurphyWeb could you please have a look at this when you get the chance?

Thank you!

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,5 @@ jspm_packages
 .node_repl_history
 
 .env
+
+.eslintrc.js
```
Comment (SimonLab) [inline .gitignore]: :+1:

File: lib/queries/getChallenge.js [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+    + 'challenges.title AS chal_title, '
+    + 'challenges.description AS chal_desc, '
+    + 'tags.name AS tags_name, '
+    + 'org_id, tags_id '
+    + 'FROM '
+    + 'challenges '
+    + 'JOIN tags_challenges ON tags_challenges.challenges_id = challenges.id '
+    + 'JOIN tags ON tags_challenges.tags_id = tags.id '
+    + 'WHERE '
+    + 'tags.active = true AND challenges.active = true '
```
Comment (SimonLab) [inline lib/queries/getChallenge.js]: ```tags.active = true``` Do we need to filter the inactive tags belonging to a challenge here or do we want to return all the tags and display them in a different format (different colour?) on the view?

File: tests/setup.test.js [hunk 3]
```diff
@@ -6,21 +6,59 @@ var initialTablesData = require('../fixtures/importMockData.js');
...
-
-test('Server start ok', function (t) {
+function setup (configData, cb) {
   initialTablesData(config, function () {
-    initServer(config, function (err, server, pool) {
-      if (err) {
-        return t.fail('Error starting the server, error: ', err);
+    initServer(config, function (error, server, pool) {
+      if (error) {
+        return cb(false);
```
Comment (SimonLab) [inline tests/setup.test.js]: I think we should return the error on the callback

File: lib/queries/getChallenge.js (hunk 2 above)
Comment (mk4111) [inline lib/queries/getChallenge.js]: @SimonLab that's a good point - we should probably ask the clients which they would prefer. I was under the assumption that if they were inactive, the clients would not want the users to see the tag at all...

File: tests/setup.test.js (hunk 3 above)
Comment (mk4111) [inline tests/setup.test.js]: oops - very true!

**Pull request merged**